bentoml/_internal/io_descriptors/json.py

```diff
@@ -269,11 +284,65 @@ async def to_http_response(
         if ctx is not None:
             res = Response(
                 json_str,
-                media_type=MIME_TYPE_JSON,
+                media_type=self._mime_type,
                 headers=ctx.response.metadata,  # type: ignore (bad starlette types)
                 status_code=ctx.response.status_code,
             )
             set_cookies(res, ctx.response.cookies)
             return res
         else:
-            return Response(json_str, media_type=MIME_TYPE_JSON)
+            return Response(json_str, media_type=self._mime_type)
+
+    async def from_proto(
+        self,
+        field: struct_pb2.Value | pb.Part | bytes,
+        *,
+        _use_internal_bytes_contents: bool = False,
+    ) -> JSONType:
+        from google.protobuf.json_format import MessageToDict
+
+        if not _use_internal_bytes_contents:
+            if isinstance(field, pb.Part):
+                field = field.json
+            assert isinstance(field, struct_pb2.Value)
+            parsed = MessageToDict(field, preserving_proto_field_name=True)
+
+            if self._pydantic_model:
+                try:
+                    return self._pydantic_model.parse_obj(parsed)
+                except pydantic.ValidationError as e:
+                    raise BadInput(f"Invalid JSON input received: {e}") from None
+        else:
+            assert isinstance(field, bytes)
+            content = field
+            if self._pydantic_model:
+                try:
+                    return self._pydantic_model.parse_raw(content)
+                except pydantic.ValidationError as e:
+                    raise BadInput(f"Invalid JSON input received: {e}") from None
+
+            try:
+                parsed = json.loads(content)
+            except json.JSONDecodeError as e:
+                raise BadInput(f"Invalid JSON input received: {e}") from None
+
+        return parsed
+
+    async def to_proto(self, obj: JSONType) -> struct_pb2.Value:
+        if LazyType["pydantic.BaseModel"]("pydantic.BaseModel").isinstance(obj):
+            obj = obj.dict()
+        msg = struct_pb2.Value()
+        # To handle None cases.
+        if obj is not None:
+            from google.protobuf.json_format import ParseDict
+
+            if isinstance(obj, (dict, str, list, float, int, bool)):
+                # ParseDict handles google.protobuf.Struct type
+                # directly if given object is of type dict
```

Comment:
And correct comment below as well. Good comments, though!

```suggestion
                # directly if given object has a supported type
```